core/templates/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.spec.ts

```diff
@@ -0,0 +1,178 @@
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+describe('Oppotunities backend API serice', function() {
+  let contributionOpportunitiesBackendApiService:
+    ContributionOpportunitiesBackendApiService = null;
+  let httpTestingController: HttpTestingController;
+  let explorationOpportunitySummaryObjectFactory:
+    ExplorationOpportunitySummaryObjectFactory = null;
+  let skillOpportunityObjectFactory:
+    SkillOpportunityObjectFactory = null;
+  let urlInterpolationService:
+    UrlInterpolationService = null;
+  const skillResponse = {
+    opportunities: [{
+      id: 'skill_id',
+      skill_description: 'A new skill for question',
+      topic_name: 'A new topic',
+      question_count: 30
+    }],
+    next_cursor: '6',
+    more: ''
+  };
+  const sampleExplorationSumary = {
+    opportunities: [{
+      id: 'exp_id',
+      topic_name: 'Topic',
+      story_title: 'A new story',
+      chapter_title: 'Introduction',
+      content_count: 100,
+      translation_counts: {
+        hi: 15
+      }
+    }],
+    next_cursor: '6',
+    more: ''
+  };
+  let sampleSkillOpportunityResponse = null;
+  let sampleTranslationOpportunityResponse = null;
+  let sampleVoiceoverOpportunityResponse = null;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
+    contributionOpportunitiesBackendApiService =
+      TestBed.get(ContributionOpportunitiesBackendApiService);
+    explorationOpportunitySummaryObjectFactory =
+      TestBed.get(ExplorationOpportunitySummaryObjectFactory);
+    httpTestingController = TestBed.get(HttpTestingController);
+    skillOpportunityObjectFactory = TestBed.get(SkillOpportunityObjectFactory);
+    urlInterpolationService = TestBed.get(UrlInterpolationService);
+    sampleSkillOpportunityResponse = [
+      skillOpportunityObjectFactory.createFromBackendDict(
+        skillResponse.opportunities[0])
+    ];
+    sampleTranslationOpportunityResponse = [
+      explorationOpportunitySummaryObjectFactory.createFromBackendDict(
+        sampleExplorationSumary.opportunities[0]
+      )
+    ];
+    sampleVoiceoverOpportunityResponse = [
+      explorationOpportunitySummaryObjectFactory.createFromBackendDict(
+        sampleExplorationSumary.opportunities[0]
+      )
+    ];
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  it('should successfully fetch the skills data',
+    fakeAsync(() => {
+      const successHandler = jasmine.createSpy('success');
+      const failHandler = jasmine.createSpy('fail');
+
+      contributionOpportunitiesBackendApiService.fetchSkillOpportunities(
+        '').then(
+        successHandler, failHandler
+      );
+      const req = httpTestingController.expectOne(
+        urlInterpolationService.interpolateUrl(
+          '/opportunitiessummaryhandler/<opportunityType>',
+          { opportunityType: 'skill' }
+        ) + `?cursor=${''}`
```

Comment:
Actually, when I first made this test, karma was not recognizing the request. It would literally show the same URL saying expected request on url but found none. Got request on sameurl but not expected one. So I changed it to templates and it worked. I tried again right now without the template, it is working... so removing the template.

Reply:
Done!